include/cantera/oneD/StFlow.h

```diff
@@ -254,6 +255,154 @@ class StFlow : public Domain1D
+        }
+    }
+
+    //! Sets the coordinate of the left control point
+    void setLeftControlPointCoordinate(double z_left) {
+        if (m_twoPointControl) {
+                m_zLeft = z_left;
+        } else {
+            throw CanteraError("StFlow::setLeftControlPointCoordinate",
+                 "Invalid operation: two-point control is not enabled.");
+        }
+    }
+
+    //! Returns the temperature at the right control point
+    double rightControlPointTemperature() const {
+        if (m_twoPointControl) {
+            if (m_zRight != Undef) {
+                return m_tRight;
+            } else {
+                throw CanteraError("StFlow::rightControlPointTemperature",
+                    "Invalid operation: right control point location is not set");
+            }
+        } else {
+            throw CanteraError("StFlow::rightControlPointTemperature",
+                 "Invalid operation: two-point control is not enabled.");
+        }
+    }
+
+    //! Returns the z-coordinate of the right control point
+    double rightControlPointCoordinate() const {
+        if (m_twoPointControl) {
+            if (m_zRight != Undef){
+                return m_zRight;
+            } else {
+                throw CanteraError("StFlow::rightControlPointCoordinate",
+                    "Invalid operation: right control point location is not set");
+            }
+        } else {
+            throw CanteraError("StFlow::rightControlPointCoordinate",
+                 "Invalid operation: two-point control is not enabled.");
+        }
+    }
+
+    //! Sets the temperature of the right control point
+    void setRightControlPointTemperature(double temperature) {
+        if (m_twoPointControl) {
+            if (m_zRight != Undef){
+                m_tRight = temperature;
+            } else {
+                throw CanteraError("StFlow::setRightControlPointTemperature",
+                    "Invalid operation: right control point location is not set");
+            }
+        } else {
+            throw CanteraError("StFlow::setRightControlPointTemperature",
+                 "Invalid operation: two-point control is not enabled.");
+        }
+    }
+
+    //! Sets the coordinate of the right control point
+    void setRightControlPointCoordinate(double z_right) {
+        if (m_twoPointControl) {
+                m_zRight = z_right;
+        } else {
+            throw CanteraError("StFlow::setRightControlPointCoordinate",
+                 "Invalid operation: two-point control is not enabled.");
+        }
+    }
+
+    //! Sets the status of the two-point control
+    void enableTwoPointControl(bool twoPointControl) {
+        if (isStrained()){
+            m_twoPointControl = twoPointControl;
+        } else {
+            throw CanteraError("StFlow::enableTwoPointControl",
+                "Invalid operation: two-point control can only be used with axisymmetric flames.");
```

Comment:
To keep this to <88 characters:
```suggestion
            throw CanteraError("StFlow::enableTwoPointControl",
                "Invalid operation: two-point control can only be used "
                "with axisymmetric flames.");
```